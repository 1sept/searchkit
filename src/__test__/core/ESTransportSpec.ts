import {ESTransport} from "../../"
import "jasmine-ajax"

describe("ESTransport", ()=> {

  beforeEach(()=> {
    jasmine.Ajax.install()
    this.host = "http://search:9200/"
    this.transport = new ESTransport(this.host)
  })

  afterEach(()=> {
    jasmine.Ajax.uninstall()
  })

  it("constructor()", ()=> {
    expect(this.transport.host).toBe(this.host)
    expect(this.transport.options.headers).toEqual({})
    let axiosConfig = this.transport.axios.defaultConfig
    expect(axiosConfig.baseURL).toBe(this.host)
    expect(axiosConfig.timeout).toBe(ESTransport.timeout)
    expect(axiosConfig.headers).toBe(this.transport.options.headers)
  })

  it("constructor() - headers", ()=> {
    const transport = new ESTransport(this.host, {
      headers:{
        "Content-Type":"application/json",
      },
      basicAuth:"key:val"
    })
    expect(transport.options.headers).toEqual({
      "Content-Type":"application/json",
      "Authorization":"Basic " + btoa("key:val")
    })
  })

  it("_search()", (done)=> {
    let mockResults = {hits:[1,2,3]}
    jasmine.Ajax.stubRequest(this.host + "_search").andReturn({
      "responseText": JSON.stringify(mockResults)
    });
    this.transport._search({
      size:10,
      from:0
    }).then((result)=> {
      expect(result.hits).toEqual([1,2,3])
      let request = jasmine.Ajax.requests.mostRecent()
      expect(request.method).toBe("POST")
      expect(request["data"]()).toEqual(
        {size:10, from:0})
      done()
    })
  })

  it("_search()", (done)=> {
    let mockResults = {responses:[1,2,3]}
    jasmine.Ajax.stubRequest(this.host + "_msearch").andReturn({
      "responseText": JSON.stringify(mockResults)
    });
    this.transport._msearch(["a", "b", "c"]).then((result)=> {
      expect(result).toEqual([1,2,3])
      let request = jasmine.Ajax.requests.mostRecent()
      expect(request.method).toBe("POST")
      expect(request["data"]()).toEqual(
        ["a", "b", "c"])
      done()
    })
  })

  it("search() - single query", (done)=> {
    spyOn(this.transport, "_search")
      .and.returnValue(Promise.resolve("singleSearchResponse"))
    let queries = ["query1"]
    this.transport.search(queries)
      .then((result)=> {
        expect(result).toEqual(["singleSearchResponse"])
        expect(this.transport._search)
        .toHaveBeenCalledWith("query1")
        done()
      })
  })

  it("search() - multiple query", (done)=> {
    spyOn(this.transport, "_msearch")
      .and.returnValue(Promise.resolve([
        "r1", "r2"
      ]))
    let queries = ["q1", "q2"]
    this.transport.search(queries)
      .then((result)=> {
        expect(result).toEqual(["r1", "r2"])
        expect(this.transport._msearch)
          .toHaveBeenCalledWith(["q1", "q2"])
        done()
      })
  })

  it("test timeout", ()=> {
    ESTransport.timeout = 10
    this.transport = new ESTransport(this.host)
    expect(this.transport.axios.defaultConfig.timeout)
      .toEqual(10)
  })

})
