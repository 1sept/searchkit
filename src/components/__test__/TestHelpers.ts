import ReactTestUtils = require('react-addons-test-utils');

export const hasClass = (inst, className)=> {
  if(ReactTestUtils.isDOMComponent(inst.node)) {
    return inst.hasClass(className)
  } else {
    try {
      let classes = inst.node.props.children.props.className
      return (' ' + classes + ' ').indexOf(' ' + className + ' ') > -1
    } catch (e){}
  }
  return false
}
