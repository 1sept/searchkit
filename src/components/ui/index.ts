export * from "./layout"
export * from "./list"
export * from "./range"
