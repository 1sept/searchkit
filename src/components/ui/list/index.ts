export * from "./ItemComponents"
export * from "./ItemListComponents"
export * from "./ListProps"
export * from "./Select"
export * from "./TagCloud"
export * from "./Toggle"
