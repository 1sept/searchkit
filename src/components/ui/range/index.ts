export * from './RangeHistogram'
export * from './RangeSlider'
