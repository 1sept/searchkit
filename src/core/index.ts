export * from "./accessors"
export * from "./query"
export * from "./react"
export * from "./state"
export * from "./support"
export * from "./Searcher"
export * from "./SearchkitManager"
export * from "./history"
export * from "./ESRequest"
export * from "./ESMultiRequest"
