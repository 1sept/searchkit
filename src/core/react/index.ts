export * from "./SearchkitComponent"
export * from "./SearchkitProvider"
export * from "./LoadingComponent"
export * from "./ReactComponentType"
export * from "./FastClick"
export * from "./RenderComponent"
export * from "./pure-render/index"
