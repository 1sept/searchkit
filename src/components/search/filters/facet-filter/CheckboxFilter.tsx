import * as React from "react";

import { FacetAccessor} from '../../../../core'
import { ItemList} from "../../../ui"
import { FacetFilter} from "./FacetFilter"

const defaults = require("lodash/defaults")
const map = require("lodash/map")
const concat = require("lodash/concat")
const assign = require("lodash/assign")

import {FacetFilterProps, FacetFilterPropTypes} from "./FacetFilterProps"

export interface CheckboxFilterProps extends FacetFilterProps{
  value:string
}
export class CheckboxFilter extends FacetFilter<CheckboxFilterProps> {

  static propTypes = defaults({
    value:React.PropTypes.string.isRequired
  },FacetFilterPropTypes.propTypes)

  static defaultProps = defaults({
    size:1,
    showMore:false
  }, FacetFilter.defaultProps)
  
  getAccessorOptions(){
    let options = super.getAccessorOptions()
    return assign(options, {include:[this.props.value]})
  }

  getItems(){
    let key = this.props.value
    return [this.accessor.getBuckets()[0] || {key, doc_count:0}]
  }

}
