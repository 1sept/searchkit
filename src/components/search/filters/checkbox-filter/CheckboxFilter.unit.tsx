import * as React from "react";;
import {mount, render} from "enzyme";
import {fastClick, hasClass, jsxToHTML, printPrettyHtml} from "../../../__test__/TestHelpers"
import {CheckboxFilter} from "./CheckboxFilter";
import {SearchkitManager, Utils} from "../../../../core";
import {Toggle, ItemComponent} from "../../../ui";
import { TermQuery } from "../../../../core";
import * as ReactTestRenderer from 'react-test-renderer'

const bem = require("bem-cn");
import * as _ from "lodash"

describe("CheckboxFilter tests", () => {
  this.createWrapper = (isEnzyme) => {
    const c = (<CheckboxFilter
      id="test id" title="test title" label="test label"
      searchkit={this.searchkit}
      filter={TermQuery("test-field", "test-value")} />)
    this.wrapper = isEnzyme ? mount(c) : ReactTestRenderer.create(c)

    this.searchkit.setResults({
      hits:{
        hits:[{_id:1, title:1},{_id:2,title:2}],
        total:2
      },
      aggregations: {
        "test id1": {
          doc_count: 50
        }
      }
    })

    this.accessor = this.searchkit.accessors.getAccessors()[0]
  }

  beforeEach(() => {
    Utils.guidCounter = 0

    this.searchkit = SearchkitManager.mock()
    this.searchkit.translateFunction = (key) => {
      return {
        "test option 1": "test option 1 translated"
      }[key]
    }

  });

  it('renders correctly', () => {
    this.createWrapper(false)
    expect(this.wrapper).toMatchSnapshot()
  });

  it('clicks options', () => {
    this.createWrapper(true)
    expect(this.accessor.state.getValue()).toEqual(null)
    let option = this.wrapper.find(".sk-item-list").children().at(0)
    fastClick(option)
    expect(hasClass(option, "is-active")).toBe(true)
    expect(this.accessor.state.getValue()).toEqual(true)
    fastClick(option)
    expect(this.accessor.state.getValue()).toEqual(false) // Back to null ?
  })

  it("should configure accessor correctly", () => {
    this.createWrapper(true)
    expect(this.accessor.key).toBe("test id")
    let options = this.accessor.options

    expect(options).toEqual({
      "id": "test id",
      "title": "test title",
      "label": "test label",
      "translations": undefined,
      "filter": TermQuery("test-field", "test-value")
    })
  })

  it("can disable", () => {
    this.createWrapper(true)
    expect(hasClass(this.wrapper.find(".sk-panel"), "is-disabled")).toBe(false)
    this.searchkit.setResults({
      hits:{ total:0 },
      aggregations: {
        "test id1": {
          doc_count: 50
        }
      }
    })
    expect(hasClass(this.wrapper.find(".sk-panel"), "is-disabled")).toBe(true)

    expect(this.accessor.state.getValue()).toEqual(null)
    let option = this.wrapper.find(".sk-item-list").children().at(0)
    fastClick(option)
    expect(this.accessor.state.getValue()).toEqual(true)

    expect(hasClass(this.wrapper.find(".sk-panel"), "is-disabled")).toBe(false)
  })
});
