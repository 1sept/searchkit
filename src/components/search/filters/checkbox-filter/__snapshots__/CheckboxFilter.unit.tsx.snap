// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxFilter tests renders correctly 1`] = `
<div
  className="sk-panel filter--test id"
>
  <div
    className="sk-panel__header"
  >
    test title
  </div>
  <div
    className="sk-panel__content"
  >
    <div
      className="sk-item-list"
      data-qa="options"
    >
      <div
        className="sk-item-list-option sk-item-list__item"
        data-key="test label"
        data-qa="option"
        onClick={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        style={undefined}
      >
        <input
          checked={false}
          className="sk-item-list-option__checkbox"
          data-qa="checkbox"
          readOnly={true}
          type="checkbox"
        />
        <div
          className="sk-item-list-option__text"
          data-qa="label"
        >
          test label
        </div>
        <div
          className="sk-item-list-option__count"
          data-qa="count"
        >
          50
        </div>
      </div>
    </div>
  </div>
</div>
`;
